quantities/parser.hpp

```diff
@@ -9,16 +9,19 @@ namespace principia {
 namespace quantities {
 namespace internal_parser {
 
-template<typename T>
-T ParseQuantity(std::string const& s);
-
-template<typename T>
-T ParseUnit(std::string const& s);
+// This function parses the following grammar:
+//   quantity ::= double quotient_unit
+//   quotient_unit       ::= [quotient_unit /] product_unit
```

Comment:
I don't really like this rule, as far as I can tell it parses `W/sr m` as `W/(sr m)`, which is confusing at best. Can we just make it
```
quotient_unit ⩴ [quotient_unit /] exponentiation_unit
```
or even, since `W/sr/m` looks very weird,
```
quotient_unit ⩴ [product_unit /] exponentiation_unit
```
we may want to extend the grammar further in many ways, but `[product_unit /] exponentiation_unit` should cover our current use cases and parse only reasonable things.